lib/components/narrative/access-leg.js

```diff
@@ -12,48 +12,58 @@ export default class AccessLeg extends Component {
     setActiveLeg: PropTypes.func,
     setActiveStep: PropTypes.func
   }
+
   _onLegClick (e, leg, index) {
     if (this.props.active) {
       this.props.setActiveLeg(null)
     } else {
       this.props.setActiveLeg(index, leg)
     }
   }
+
   _onStepClick (e, step, index) {
     if (index === this.props.activeStep) {
       this.props.setActiveStep(null)
     } else {
       this.props.setActiveStep(index, step)
     }
   }
+
   render () {
     const { active, activeStep, index, leg } = this.props
+    const legClickHandler = (e) => this._onLegClick(e, leg, index)
```

Comment:
I'm a little confused on how to do that since it's in a loop where leg and index are changed.  Wouldn't a function need to get created each time for this to work?